units/find_outdated_rollups.go

```diff
@@ -0,0 +1,168 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/cedar/perf"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+const (
+	findOutdatedRollupsJobName = "find-outdated-rollups"
+)
+
+type findOutdatedRollupsJob struct {
+	RollupTypes []string `bson:"rollup_types" json:"rollup_types" yaml:"rollup_types"`
+
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+	env      cedar.Environment
+	seenIDs  map[string]bool
+}
+
+func init() {
+	registry.AddJobType(findOutdatedRollupsJobName, func() amboy.Job { return makeFindOutdatedRollupsJob() })
+}
+
+func makeFindOutdatedRollupsJob() *findOutdatedRollupsJob {
+	j := &findOutdatedRollupsJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    findOutdatedRollupsJobName,
+				Version: 1,
+			},
+		},
+	}
+
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func (j *findOutdatedRollupsJob) validate() error {
+	if len(j.RollupTypes) == 0 {
+		return errors.New("no rollup factories given")
+	}
+
+	return nil
+}
+
+func NewFindOutdatedRollupsJob(factories []perf.RollupFactory) (amboy.Job, error) {
+	j := makeFindOutdatedRollupsJob()
+
+	j.RollupTypes = []string{}
+	for _, factory := range factories {
+		j.RollupTypes = append(j.RollupTypes, factory.Type())
+	}
+
+	if err := j.validate(); err != nil {
+		return nil, errors.Wrap(err, "failed to create new ftdc rollups job")
+	}
+
+	j.SetID(fmt.Sprintf("find-outdated-rollups.%s", time.Now()))
+
+	return j, nil
+}
+
+func (j *findOutdatedRollupsJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	if j.env == nil {
+		j.env = cedar.GetEnvironment()
+	}
+
+	if j.seenIDs == nil {
+		j.seenIDs = map[string]bool{}
+	}
+
+	factories := []perf.RollupFactory{}
+	for _, t := range j.RollupTypes {
+		factory := perf.RollupFactoryFromType(t)
+		if factory == nil {
+			err := errors.Errorf("problem resolving rollup factory type %s", t)
+			grip.Warning(err)
+			j.AddError(err)
+			continue
+		}
+		factories = append(factories, factory)
+	}
+
+	results := model.PerformanceResults{}
+	results.Setup(j.env)
+	for i, factory := range factories {
+		for _, name := range factory.Names() {
+			after := time.Now().Add(-3 * 24 * time.Hour)
+			if err := results.FindOutdatedRollups(name, factory.Version(), after); err != nil {
+				err = errors.Wrapf(err, "problem checking for outdated rollups for %s", name)
+				grip.Warning(err)
+				j.AddError(err)
+				continue
+			}
+
+			for _, result := range results.Results {
+				if _, ok := j.seenIDs[result.Info.ID()]; !ok {
+					j.createFTDCRollupsJobs(factories[i:], result)
```

Comment:
why is this factories[i:] and not factories[i]

Reply:
It needs to check the rest of the rollups, this way there is only ever one job created per result (and thus one s3 download). So the first line in `createFTDCRollupsJobs` calls `findOutdatedFromResult` to do the check.